src/main/java/apartment/domain/ReservationService.java

```diff
@@ -0,0 +1,173 @@
+package apartment.domain;
+
+import apartment.data.DataException;
+import apartment.data.GuestRepository;
+import apartment.data.HostRepository;
+import apartment.data.ReservationRepository;
+import apartment.models.Guest;
+import apartment.models.Host;
+import apartment.models.Reservation;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class ReservationService {
+    private final ReservationRepository reservationRepository;
+    private final GuestRepository guestRepository;
+    private final HostRepository hostRepository;
+
+    public ReservationService(ReservationRepository reservationRepository, GuestRepository guestRepository, HostRepository hostRepository) {
+        this.reservationRepository = reservationRepository;
+        this.guestRepository = guestRepository;
+        this.hostRepository = hostRepository;
+    }
+
+    public List<Reservation> getReservationForParticularHost(Host host) throws DataException {
+        List<Reservation> reservations = reservationRepository.findByHostId(host.getId());
+
+        Map<Integer,Guest> guestMap = guestRepository.findAll().stream()
+                .collect(Collectors.toMap(guest -> guest.getId(),guest -> guest));
+
+
+
+        if (reservations.size() != 0) {
+            reservations.stream().forEach(reservation -> reservation.setGuest(guestMap.get(reservation.getGuestId())));
+            reservations.forEach(reservation -> reservation.setHost(host));
+        }
+
+        reservations = reservations.stream().sorted((r1, r2) -> r1.getStartDate().compareTo(r2.getStartDate())).collect(Collectors.toList());
+
+        return reservations;
+    }
+
+
+    public boolean removeReservationById(int reservationId, Host host) throws DataException {
+        List<Reservation> reservations = reservationRepository.findByHostId(host.getId());
+        Reservation reservation = reservations.stream().filter(r -> r.getId() == reservationId).findFirst().orElse(null);
+        if (reservation == null) {
+            return false;
+        }
+        if (reservation.getStartDate().compareTo(LocalDate.now()) <= 0) {
+            return false;
+        }
+
+
+        return reservationRepository.removeById(reservationId, host);
+    }
+
+    public Result<Boolean> updateReservation(Reservation updatedReservation) throws DataException {
+        Result<Reservation> validationResults = validate(updatedReservation, true);
+
+        Result<Boolean> output = new Result<>();
+        output.setMessage(validationResults.getErrorMessages());
```

Comment:
I don't see any reason why you can't just use the above `validationResults` variable for the results to return from this method.